### PR TITLE
Enables custom default tags and ability to add no tags (issue #51)

### PR DESCRIPTION
**Pull request by formsandlines**
The default tag will still remain `"clojure"`, but users can change it by adding `:default-metadata {:tags […]}` to their `opts` map in bb.edn, which also enables them to add multiple default tags or none at all (with an empty vector). I hope my changes are aligned with the intended use of `:default-metadata`.

For a single post, users can just add `--tags` without any arguments to omit tags even if the default vector is not empty. Maybe there could also be an option `--no-tags` for clarity of intent, but I don’t think it’s necessary.

### CODE REVIEW

File: src/quickblog/api.clj [hunk 1]
```diff
@@ -489,9 +489,11 @@
       :ref "<tags>"
       :coerce []}}}}
   [opts]
-  (let [{:keys [file title posts-dir tags]
+  (let [{:keys [file title posts-dir tags default-metadata]
          :as opts} (apply-default-opts opts)
-        tags (if (empty? tags) ["clojure"] tags)]
+        tags (cond (empty? tags)   (:tags default-metadata)
+                   (= tags [true]) [] ;; `--tags` without arguments
```
Comment (borkdude) [inline src/quickblog/api.clj:495]: Could you try bumping babashka cli? This behavior might have been "fixed" in babashka CLI

**Comment by borkdude** [comment]
I should enable a test runner on CI, but for now you can run tests locally with `bb test`. If possible, it would be good to add a test for this PR as well.

**Comment by formsandlines** [comment]
Oh well, seems like the test for `migrate` fails, because tags are now ordered before the post title in the document after calling `api/migrate`. I don’t know what causes this, but I’ll see if I can fix it.

I only tested the behavior of the `new` command with various tags/defaults in a separate project, but will try to add tests as well, if I can figure out how to set them up correctly.

File: src/quickblog/api.clj (hunk 1 above)
Comment (formsandlines) [reply src/quickblog/api.clj:495]: Which version are you referring to? The most recent version on Clojars is 0.6.41, which is the version that is already in deps.edn.

Do you mean the behavior that a CLI option without arguments returns `true`, which I tried to match here?

**Comment by borkdude** [comment]
I think you can change the test by splitting the lines and comparing the set of both lines.

**Comment by formsandlines** [comment]
Just changed the failing test according to your suggestion, which I hope I understood correctly. I wasn’t sure what exactly “migrate” does and if the order is important.

**Comment by borkdude** [comment]
Thanks and congrats with your first merged PR :)

**Comment by formsandlines** [comment]
Glad I could help, had to learn some new things about Git and GitHub along the way. :D

**Pull request merged**